src/test/controllerscriptenginelegacy_test.cpp

```diff
@@ -745,3 +745,34 @@ TEST_F(ControllerScriptEngineLegacyTest, screenWillSentRawDataIfConfigured) {
     ASSERT_ALL_EXPECTED_MSG();
 }
 #endif
+
+TEST_F(ControllerScriptEngineLegacyTest, convertCharsetUndefinedOnUnknownCharset) {
+    const auto result = evaluate("engine.convertCharset('NULL', 'Hello!')");
+    EXPECT_EQ(qjsvalue_cast<QByteArray>(result), QByteArray());
+}
+
+TEST_F(ControllerScriptEngineLegacyTest, convertCharsetCorrectValueWellKnown) {
+    const auto result = evaluate(
+            "`${engine.convertCharset(engine.WellKnownCharsets.LATIN_9, 'Hello!')}`");
```

Comment:
I think the implicit `toString` cast is wrong I think (and likely also the cause for the failing test).
```suggestion
    const auto result = evaluate(
            "engine.convertCharset(engine.WellKnownCharsets.LATIN_9, 'Hello!'));
```